pkg/cmd/logger.go

```diff
@@ -36,6 +36,15 @@ func NewLogger() log.Logger {
 	return cli.NewLogger(writer, 0)
 }
 
+// NewLoggerWithWriter returns the standard logger used by the kind CLI
+// This logger writes to given external writter
+func NewLoggerWithWriter(writer io.Writer) log.Logger {
+	if env.IsSmartTerminal(writer) {
+		writer = cli.NewSpinner(writer)
+	}
+	return cli.NewLogger(writer, 0)
```

Comment:
what writer do you intend to pass to this?

you can already implement the logger interface, this particular implemenation makes assumptions about writing to a console :thinking:

Reply:
This is something for automation and for passing the logs to UI during the automation.

In my case the writer is custom go writer which is listening on a channel to read the content and pass the content to the UI through web-sockets.

Let me know if you need more details about the use-case. Writer is smart enough to handle the logs written to console.

Reply:
I got it. Thanks for the review. I use the logger interface to handle my scenario as you mentioned.

Thanks for the time.

Reply:
i think the logger interface is indeed the way to go here 😅 
thanks